docs/frontend-system/architecture/03-extensions.md

```diff
@@ -85,68 +85,292 @@ Note that while the `createExtension` is public API and used in many places, it
+With creating an extension by using `createExtension(...)` you have the advantage that the extension can be anything in your Backstage application. We realised that this comes with the trade-off of having to repeat boilerplate code for similar building blocks. Here extension creators come into play for covering common building blocks in Backstage like pages using `createPageExtension`, themes using the `createThemeExtension` or items for the navigation using `createNavItemExtension`.
 
- - One example of a YAML configurations for the above extension
+If we follow the example from above all items of the navigation have similarites, like they all want to be attached to the same extension with the same input as well as rendering the same navigation item component. Therefor `createExtension` can be abstracted for this use case to `createNavItemExtension` and if we add the extension to the app it will end up in the right place & looks like we expect a navigation item to look.
 
- -->
+```tsx
+export const HomeNavIcon = createNavItemExtension({
+  routeRef: routeRefForTheHomePage,
+  title: 'Home',
+  icon: HomeIcon,
+});
+```
 
-## Extension Creators
+### Extension Kind
 
-<!--
+With the example `HomeNavIcon` will end up on the extension input `items` of the extensions with the id `app/nav`. It raises the question what the `id` of the `HomeNavIcon` itself is. The extension creator for the navigation item has a defined `kind`, which by convention matches the own name. So in this example `createNavItemExtension` sets the kind to `nav-item`.
+
+The `id` of the extension is then build out of `namespace`, `name` & `kind` like the following - where `namespace` & `name` are optional properties that can be passed to the extension creator:
 
-Introduce the concept of extension creators, talk about how they are used to provide an opinionated way of creating extensions for specific use cases. Mention a few of the built-in creators, link to the reference section on built-in extension kinds (TBD).
+```
+id: kind:namespace/name
+```
 
- - Example of using an existing extension creator to create an extension
+For more information on naming of extension refer to the naming patterns documentation (TODO: Link)
 
-Explain that each extension creator creates a specific "kind" of extensions - introduce the concept of extension kinds. Show an example of how the kind is used to build up the ID.
+### Extension Creators in libraries
 
- - Example of creating a new extension creator for creating a specific kind of extension
+Extension creators should be exported from library packages (e.g. `*-react`, `*-common`) rather than plugin packages.
+
+<!--
 
 Explain that extension creators should be exported from library packages (`-react`) rather than plugin packages.
 
+philipph: What is the advantage - not super obvious to me from the comment. We would also need to change this e.g. for the catalog as it currently is exported from `catalog` alpha directly.
```

Comment:
It's a bit more nuanced that what the comment implies. Extension creators that are intended for use by other plugins or modules should be exported from the `-react` package. Never `-common` though since we can't have web-only things there.

There are situations where exporting from the plugin package itself might be alright, primarily that's for when a particular extension kind is really only intended as an internal customization. Still, if a plugin has a `-react` package it's usually best to place the creators there, and yep the catalog filter creator should be moved over imo.

We should probably say something to the effect of "if you only intend for an extension kind to be used for internal customizations and not by other open source plugins, then you can place the extension creator in the plugin package itself. But if you already have a `-react` package or intend for the extension creator to be used by other open source plugins, then you should always put extension creators in a `-react` package.